src/emailcontactmodel.cpp

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (c) 2020- 2021 Open Mobile Platform LLC.
+ *
+ * This program is licensed under the terms and conditions of the
+ * Apache License, version 2.0.  The full text of the Apache License is at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+#include "emailcontactmodel.h"
+#include <QHash>
+
+
+namespace  {
+    const int MAX_COUNT = 100;
+}
+
+
+EmailContactModel::EmailContactModel(QAbstractItemModel *parent)
+    : QAbstractListModel(parent)
+{
+    initialize(MAX_COUNT);
+}
+
+
+static void push(QHash<QMailAddress, QMailTimeStamp> &container, const QMailAddress &address, const QMailTimeStamp &stamp)
+{
+    if (!address.isEmailAddress())
+        return;
+
+    QHash<QMailAddress, QMailTimeStamp>::iterator i = container.find(address);
+    if (i == container.end()) {
+        container.insert(address, stamp);
+    } else {
+        if (i.value().toLocalTime() < stamp.toLocalTime())
+            i.value() = stamp;
+    }
+}
+
+
```

Comment:
Minor nitpicking, but commonly the separation is just one line :)

Reply:
done